app/bundles/EmailBundle/Swiftmailer/Amazon/AmazonCallback.php

```diff
@@ -51,12 +51,16 @@ public function processCallbackRequest(Request $request)
             throw new HttpException(400, 'AmazonCallback: Invalid JSON Payload');
         }
 
-        if (!isset($payload['Type']) && !isset($payload['eventType'])) {
-            throw new HttpException(400, "Key 'Type' not found in payload ");
-        }
-
         // determine correct key for message type (global or via ConfigurationSet)
-        $type = (array_key_exists('Type', $payload) ? $payload['Type'] : $payload['eventType']);
+        if (isset($payload['Type'])) {
+            $type = $payload['Type'];
+        } elseif (isset($payload['eventType'])) {
+            $type = $payload['eventType'];
+        } elseif (isset($payload['notificationType'])) {
```

Comment:
@mabumusa1 I'm talking about this change. Before this PR and in your plugin, only the `Type' and `eventType' headers are checked. Are complaints correctly recognised in your production environment without this change?

Reply:
@mabumusa1 You're right, so if you've already implemented this feature, I think we can close this PR, right?

Reply:
@patrykgruszka we can close this one, please feel free to use the code changes on the plugin. I will release it with M5